grammars/prql-lezer/src/prql.grammar

```diff
@@ -3,114 +3,153 @@
-  op_unary { "-" | "+" | "==" }
+  //op_bin_only { "*" | "/" | "//" | "%" | "!=" | ">=" | "<=" | "~=" | ">" | "<" | "??" | "&&" | "||" }
+  //op_unary { "-" | "+" | "==" }
   end { @eof }
-  line_wrap { "\\" }
-  wrapped_line { newline+ (Comment newline+)* line_wrap }
+  lineWrap { "\\" }
+  wrappedLine { newline+ (Comment newline+)* lineWrap }
   newline { "\n" }
   // TODO: Because this can also be used to compile to BETWEEN, ranges should
   // allow any literal, and arguably any expression.
-  Range { @digit+ ".." @digit+ }
+  RangeExpression { @digit+ ".." @digit+ }
   // Couldn't manage to do these & the interpolated as a template; couldn't
   // find how to negate a variable template
-  String { $["] !["]* $["] | $['] ![']* $['] }
+  String { '"""' ![Z]* '"""' | "'''" ![Z]* "'''" | $["] !["]* $["] | $['] ![']* $['] }
 
   // TODO: not getting the interpolations highlighted; it just shows the whole
   // string as a string, because these are all within the `@tokens` block. But
   // they need to be within this block, because it's not possible to have
   // negations (e.g. `![{'}` outside it.
-  String_interpolated_single { $['] ( ![{'] | Interpolation_inner_single )* $['] }
-  String_interpolated_double { $["] ( ![{"] | Interpolation_inner_double )* $["] }
+  stringInterpolatedSingle { $['] ( ![{'] | interpolationInnerSingle )* $['] }
+  stringInterpolatedDouble { $["] ( ![{"] | interpolationInnerDouble )* $["] }
 
-  Interpolation_inner_single { "{" ![}']* "}" }
-  Interpolation_inner_double { "{" ![}"]* "}" }
+  interpolationInnerSingle { "{" ![}']* "}" }
+  interpolationInnerDouble { "{" ![}"]* "}" }
 
-  Interpolated_string<prefix> { prefix ( String_interpolated_double | String_interpolated_single ) }
+  interpolatedString<prefix> { prefix ( stringInterpolatedDouble | stringInterpolatedSingle ) }
 
-  R_string {Interpolated_string<'r'>}
-  S_string {Interpolated_string<'s'>}
-  F_string {Interpolated_string<'f'>}
+  R_string { interpolatedString<'r'> }
+  S_string { interpolatedString<'s'> }
+  F_string { interpolatedString<'f'> }
 
   // We need to give precedence to `Op_bin` so we don't get `x+y` as `x` & `+y`.
   // R, S & F strings have precedence over idents beginning with r / s / f (we could
   // use specialize but I think means we need to redefine `String` for each)
-  @precedence { R_string, S_string, F_string, op_bin_only, ident_part }
-
-  @precedence { Range, Float, Integer, time_unit, ident_part }
+  @precedence { RangeExpression, R_string, S_string, F_string, Float, TimeUnit, Integer, identPart, identifier, Docblock, Comment }
 }
 
-Def { @specialize<ident_part, "let"> ident_part "=" (Nested_pipeline (newline+ | end) | Lambda) }
-
-Lambda { Lambda_param* "->" expr_call ( newline+ | end ) }
-Type_def { "<" Type_term ( "|" Type_term)* ">" }
-Type_term { ident_part Type_def? }
-Lambda_param { ident_part Type_def? (":" Expr)? }
-
 @external propSource prqlHighlight from "./highlight"
+
+@detectDelim
```

Comment:
What does this do?

Reply:
It is used in most of the grammars, it is used in the CSS, Java, JavaScript, JSON, PHP, Python, Rust, Sass and XML grammars.

The documentation states:
> If you put a @detectDelim directive at the top level of your grammar file, the parser generator will automatically detect rule delimiter tokens, and create [openedBy](https://lezer.codemirror.net/docs/ref/#common.NodeProp%5EopenedBy) and [closedBy](https://lezer.codemirror.net/docs/ref/#common.NodeProp%5EclosedBy) props for rules where it finds them.

and...
> And finally, the parser generator can be asked to automatically infer matching delimiters with a @detectDelim directive. This will cause it to add [metadata](https://lezer.codemirror.net/docs/ref/#tree.NodeProp%5EclosedBy) to those node types, which the editor can use for things like bracket matching and automatic indentation.

Reply:
Ah thanks! 

(and fine to just tell me to look at the docs in future!)